helper/resource/testing.go

```diff
@@ -1036,6 +1033,41 @@ func ComposeAggregateTestCheckFunc(fs ...TestCheckFunc) TestCheckFunc {
 // attributes using the special key syntax, checking a list, map, or set
 // attribute directly is not supported. Use TestCheckResourceAttr with
 // the special .# or .% key syntax for those situations instead.
+//
+// The following is an example of using [ExpectKnownValue] with [knownvalue.NotNull]
+// to replicate the behaviour of TestCheckResourceAttrSet.
```

Comment:
I was thinking that the examples might serve for "organic" adoption of, or experimentation with, the new state checks rather than having an overt declaration that the `TestCheckFunc` implementations are legacy, or will be deprecated until such time as we're happy to move the new plan and state checks out of experimental status and have all of the alternative implementations for replicating built-in `TestCheckFunc`(s) in place. But I'm happy to add some additional verbiage if that's the consensus.

Reply:
I lean a little towards Austin's clarity request here, it does feel like there needs to be something to describe why the example is there. Maybe something like "An experimental interface exists to to potentially replace this functionality in the future and feedback would be appreciated. This example performs the same check with that experimental interface:"

Reply:
Fair enough. Thanks for the suggested wording, have incorporated throughout.